### PR TITLE
.*: spilt profile info from `Interpreters::Join` and implement `PipelineExecutor::getRuntimeStatistics`

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: ref #6518 

Problem Summary:

### What is changed and how it works?
- Let DAGContext no longer hold JoinPtr, so that JoinPtr can be destroyed after build/probe is executed, freeing memory from https://github.com/pingcap/tiflash/pull/7750

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/Executor/PipelineExecutor.cpp [hunk 1]
```diff
@@ -139,13 +139,18 @@ RU PipelineExecutor::collectRequestUnit()
...
 {
+    assert(root_pipeline);
     return root_pipeline->getSampleBlock();
 }
 
 BaseRuntimeStatistics PipelineExecutor::getRuntimeStatistics() const
 {
-    // TODO support getRuntimeStatistics
+    assert(root_pipeline);
+    auto final_plan_exec_id = root_pipeline->getFinalPlanExecId();
```
Comment (yibin87) [inline dbms/src/Flash/Executor/PipelineExecutor.cpp:149]: Do we need to check not "" exec_id here?

**Review by yibin87 — APPROVED** [review]
Others LGTM

File: dbms/src/Flash/Executor/PipelineExecutor.cpp [hunk 2]
```diff
@@ -139,13 +139,18 @@ RU PipelineExecutor::collectRequestUnit()
...
     BaseRuntimeStatistics runtime_statistics;
+    const auto & final_profile_infos = context.getDAGContext()->getOperatorProfileInfosMap()[final_plan_exec_id];
+    for (const auto & profile_info : final_profile_infos)
+        runtime_statistics.append(*profile_info);
```
Comment (SeaRise) [inline dbms/src/Flash/Executor/PipelineExecutor.cpp:153]:
```suggestion
    if (!final_plan_exec_id.empty())
    {
        const auto & final_profile_infos = context.getDAGContext()->getOperatorProfileInfosMap()[final_plan_exec_id];
        for (const auto & profile_info : final_profile_infos)
            runtime_statistics.append(*profile_info);
    }
```

File: dbms/src/Flash/Executor/PipelineExecutor.cpp (hunk 1 above)
Comment (SeaRise) [reply dbms/src/Flash/Executor/PipelineExecutor.cpp:149]: ok, updated.

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**